### ISSUE TITLE
Https TLS only strange behaviour

### ISSUE THREAD

**Issue by GoogleCodeExporter** [nilsbraden/ttrss-reader-fork#233]
```
What steps will reproduce the problem?
1. Try to connect with https
2. Server-side allow only TLS because of SNI
3. use non-self-signed certicate

What is the expected output? What do you see instead?
Normal usage like with http. Instead every connect to server i get the message 
that the connection failed with javax.net.ssl.SSLProtocolException: Handshake 
failed. But in the serverlogs i see the connection established on https.

What version of the product are you using? On what operating system? And
what server-version is your TTRSS-server running?
App: 1.73
Ttrss: 1.10


Please provide any additional information below.
If you want, i will try to upload a screenshot, but at the moment Chrome says 
it can't open it.
```

Original issue reported on code.google.com by `pinc...@googlemail.com` on 4 Feb 2014 at 1:07

**Comment by wiewo** [comment]
Hi,
I have a very similar problem. The app connects to my server, starts syncing and after a random time it aborts with the attached error.
![screenshot](https://cloud.githubusercontent.com/assets/9152142/7182844/34ca378c-e454-11e4-8aea-36cfdb7e2451.png)
Sometimes the error happens every second, sometimes it syncs a few feeds and then there are some up to date, while the rest does not get updated.
In my apache config I have the following settings:
```
Header add Strict-Transport-Security "max-age=15768000"
SSLCompression off
SSLProtocol all -SSLv2 -SSLv3 
SSLHonorCipherOrder on
SSLCipherSuite "EECDH+ECDSA+AESGCM EECDH+aRSA+AESGCM EECDH+ECDSA+SHA384 EECDH+ECDSA+SHA256 EECDH+aRSA+SHA384 EECDH+aRSA+SHA256 EECDH !aNULL !eNULL !LOW !3DES !MD5 !EXP !PSK !SRP !DSS !RC4"
```
The server uses a self-signed certificate. 
The error also happens with the current version from master.  I played a bit with the code in ttrssreader/src/main/java/org/ttrssreader/net/SSLSocketFactoryEx.java and altered the prefered ciphers, protocols etc., but no luck so far... 
Maybe the error gets triggered by a timeout?

**Comment by nilsbraden** [comment]
Is this still an issue?

**Comment by wiewo** [comment]
In the meantime I figured out that the error got in fact triggered by many timeouts. I had too many feeds on my server and the server could not handle the load. As soon as I had purged a lot of stuff, the error disappeared. Currently, I am running a patched version of Replicant where I removed the internal fallback of Android to SSLv3. With this version, the fallback error does not happen, even with the timeouts.
So this is not an issue anymore.
Thanks!
